hibernate-core/src/test/java/org/hibernate/jpa/test/criteria/alias/CriteriaMultiselectAliasTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.jpa.test.criteria.alias;
+
+import java.util.List;
+import java.util.Map;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Tuple;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+import org.hibernate.query.Query;
+import org.hibernate.transform.Transformers;
+
+import org.hibernate.testing.jdbc.SQLStatementInterceptor;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hibernate.testing.transaction.TransactionUtil.doInJPA;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Vlad Mihalcea
+ */
+public class CriteriaMultiselectAliasTest extends BaseEntityManagerFunctionalTestCase {
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] {
+			Book.class
+		};
+	}
+
+	@Before
+	public void init() {
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			Book book = new Book();
+			book.id = 1;
+			book.name = bookName();
+
+			entityManager.persist( book );
+		} );
+	}
+
+	@Test
+	public void test() throws Exception {
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			final CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+
+			final CriteriaQuery<Object[]> query = cb.createQuery( Object[].class );
+
+			final Root<Book> entity = query.from( Book.class );
+			query.multiselect(
+					entity.get( "id" ).alias( "id" ),
+					entity.get( "name" ).alias( "name" )
```

Comment:
Wondering if maybe you should use aliases different from the column names? That would make the test more natural.

Reply:
Makes sense. I'll push a new change. Maybe it's worth using a `StringBuilder` as well for the path building logic.